src/Symfony/Component/RateLimiter/Policy/FixedWindowLimiter.php

```diff
@@ -69,15 +69,17 @@ public function reserve(int $tokens = 1, float $maxTime = null): Reservation
                 $reservation = new Reservation($now, new RateLimit($window->getAvailableTokens($now), \DateTimeImmutable::createFromFormat('U', floor($now)), true, $this->limit));
             } else {
                 $waitDuration = $window->calculateTimeForTokens($tokens);
+                $timerDate = \DateTimeImmutable::createFromFormat('U.u', $window->getTimer());
+                $retryAfterDate = $timerDate->modify('+'.ceil($window->getExpirationTime()).' seconds');
 
                 if (null !== $maxTime && $waitDuration > $maxTime) {
                     // process needs to wait longer than set interval
-                    throw new MaxWaitDurationExceededException(sprintf('The rate limiter wait time ("%d" seconds) is longer than the provided maximum time ("%d" seconds).', $waitDuration, $maxTime), new RateLimit($window->getAvailableTokens($now), \DateTimeImmutable::createFromFormat('U', floor($now + $waitDuration)), false, $this->limit));
+                    throw new MaxWaitDurationExceededException(sprintf('The rate limiter wait time ("%d" seconds) is longer than the provided maximum time ("%d" seconds).', $waitDuration, $maxTime), new RateLimit($window->getAvailableTokens($now), $retryAfterDate, false, $this->limit));
                 }
 
                 $window->add($tokens, $now);
 
-                $reservation = new Reservation($now + $waitDuration, new RateLimit($window->getAvailableTokens($now), \DateTimeImmutable::createFromFormat('U', floor($now + $waitDuration)), false, $this->limit));
+                $reservation = new Reservation($now + $waitDuration, new RateLimit($window->getAvailableTokens($now), $retryAfterDate, false, $this->limit));
```

Comment:
Previously `Reservation(timeToAct: )` and `RateLimit(retryAfter: )` parameters were virtually the same (int/float rounding inconsistent, but both coming from `$now + $waitDuration`).

After the change`Reservation` would wait for `$waitDuration` (which is effectively `$this->interval` seconds from now), while `RateLimit` would wait for `$window->getExpirationTime()` seconds (effectively `$this->interval`) from `$window->getTimer()` - the inconsistency is potentially even bigger than before.